File: 1967_number-of-strings-that-appear-as-substrings-in-word/csharp/main.cs
using System;
using System.Text.Json;

public class Program
{
    public static void Main()
    {
        var lines = Console.In.ReadToEnd().Split('\n');
        string[] patterns = JsonSerializer.Deserialize<string[]>(lines[0]);
        string word = JsonSerializer.Deserialize<string>(lines[1]);
        var result = new Solution().NumOfStrings(patterns, word);
        Console.WriteLine(JsonSerializer.Serialize(result));
    }
}

File: 1967_number-of-strings-that-appear-as-substrings-in-word/rust/main.rs
use std::io::{self, Read};

mod solution;
use solution::Solution;

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let data: Vec<&str> = input.lines().collect();
    let patterns: Vec<String> = serde_json::from_str(data[0]).unwrap();
    let word: String = serde_json::from_str(data[1]).unwrap();
    let result = Solution::num_of_strings(patterns, word);
    println!("{}", serde_json::to_string(&result).unwrap());
}
